source/common/config/utility.h

```diff
@@ -229,6 +229,15 @@ class Utility {
    */
   static std::string resourceName(const ProtobufWkt::Any& resource);
 
+  /**
+   * Envoy can have both pre-defined tag names in well_known_names and user
+   * provided tag names through bootstrap configuration. Check the confilict of
+   * all the tag names to avoid unexpected tag name overwriting. If found a
+   * confilict, throw an exception.
+   * @param bootstrap bootstrap proto.
+   */
```

Comment:
It might not be consistently applied across the Envoy code base but this is a good opportunity to use the `@throws` doxygen tag to specify what exception type and when it is thrown.